pkgs/top-level/all-packages.nix

```diff
@@ -38874,4 +38874,13 @@ with pkgs;
   dillo = callPackage ../by-name/di/dillo/package.nix {
     fltk = fltk13;
   };
+
+  dnspy = callPackage ../by-name/dn/dnspy/package.nix {
+    wine = wine64Packages.staging;
+  };
+
+  dnspy32 = callPackage ../by-name/dn/dnspy/package.nix {
+    wine = winePackages.staging;
+    is32bit = true;
+  };
```

Comment:
Why do we have both 32 and 64 bit versions?

Reply:
I was just curious.  it sounds like there's a good reason for it.

As for the `by-name` thing:

- what does it mean to have `dnspy` in all-packages as well as by-name?  I don't think I've seen that before.  which takes precedence?

- I'd probably just add `by-name/dn/dnspy32/package.nix` and use `override` in it

Reply:
I guess logically all-packages takes precedence.  And I see why it would be tricky to do this in by-name with the way `wine` is handled.

It's probably better for it not to be in by-name if it's going to be like this.

Reply:
> what does it mean to have dnspy in all-packages as well as by-name? I don't think I've seen that before

A lot of packages do that already[^1] and it's documented in [by-name docs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md#changing-implicit-attribute-defaults).

[^1]: ![image](https://github.com/user-attachments/assets/dcdad1d8-9a55-4243-91e3-8b31fce21b79)